lib/ui/chat/chat_screen.dart

```diff
@@ -0,0 +1,273 @@
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+import 'package:go_router/go_router.dart';
+import 'package:supa_carbon_icons/supa_carbon_icons.dart';
+import 'package:whitenoise/domain/models/message_model.dart';
+import 'package:whitenoise/ui/chat/widgets/chat_input.dart';
+import 'package:whitenoise/ui/chat/widgets/contact_info.dart';
+import 'package:whitenoise/ui/chat/widgets/message_widget.dart';
+import 'package:whitenoise/ui/chat/widgets/reaction/reaction_default_data.dart';
+import 'package:whitenoise/ui/chat/widgets/reaction/reaction_hero_dialog_route.dart';
+import 'package:whitenoise/ui/chat/widgets/reaction/reactions_dialog_widget.dart';
+import 'package:whitenoise/ui/chat/widgets/status_message_item_widget.dart';
+import '../../routing/routes.dart';
+import '../core/themes/assets.dart';
+import '../core/themes/colors.dart';
+import '../../domain/dummy_data/dummy_messages.dart';
+
+class ChatScreen extends StatefulWidget {
+  const ChatScreen({super.key});
+
+  @override
+  State<ChatScreen> createState() => _ChatScreenState();
+}
+
+class _ChatScreenState extends State<ChatScreen> {
+  void showEmojiBottomSheet({
+    required MessageModel message,
+  }) {
+    showModalBottomSheet(
```

Comment:
There is a customBottomSheet class in ui/core/ui that you can reuse. You can also customise it.